plugins/src/main/java/org/smartregister/fhir/gateway/plugins/LocationHierarchyEndpointHelper.java

```diff
@@ -111,4 +115,43 @@ public List<Location> descendants(String locationId, Location parentLocation) {
         }
         return location;
     }
+
+    public Bundle getPaginatedLocations(HttpServletRequest request) {
+        String identifier = request.getParameter(Constants.IDENTIFIER);
+        String pageSize = request.getParameter(Constants.PAGINATION_PAGE_SIZE);
+        String pageNumber = request.getParameter(Constants.PAGINATION_PAGE_NUMBER);
+        Map<String, String[]> parameters = new HashMap<>(request.getParameterMap());
+
+        int count =
+                pageSize != null
+                        ? Integer.parseInt(pageSize)
+                        : Constants.PAGINATION_DEFAULT_PAGE_SIZE;
+        int page =
+                pageNumber != null
+                        ? Integer.parseInt(pageNumber)
+                        : Constants.PAGINATION_DEFAULT_PAGE_NUMBER;
+
+        int start = Math.max(0, (page - 1)) * count;
+        Location parentLocation = getLocationById(identifier);
+        List<Location> locations = getDescendants(identifier, parentLocation);
```

Comment:
Should we consider caching this?

Reply:
Yeah I think this is a good idea since we are caching the call to `getLocationHierarchy` for some great optimization. 

Because this is variation of the endpoint is paginated, you may want to add that(pagination) info as part of the key e.g. `<locationid>-<pageNo>`. This will probably only need to apply when the user has provided pagination